### PR TITLE
update to go 1.22

### PR DESCRIPTION
**Pull request by VihasMakwana**
`beats` and `elastic-agent` are already on go 1.22. Update to 1.2.

This also fixes [CI failure](https://buildkite.com/elastic/elastic-agent-system-metrics/builds/438#0192d318-7456-49d1-8ccb-bf9f1fd6febb)

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.21.13
+1.22.0
```
Comment (belimawr) [inline .go-version:1]: The `.go-version` from Beats and Elastic-Agent is `1.22.8`, so I suggest using the same here.
```suggestion
1.22.8
```

**Comment by rdner** [comment]
@rowlandgeoff should we setup the automation here too, like we have in other repositories?

**Comment by rdner** [comment]
@VihasMakwana please update versions in this file too https://github.com/elastic/elastic-agent-system-metrics/blob/main/.golangci.yml

**Comment by rowlandgeoff** [comment]
> @rowlandgeoff should we setup the automation here too, like we have in other repositories?

We chatted about this a bit in the Eng Prod office hours meeting today. The summary was that we definitely need more consistency in how we do version bumps across the repos. Various repos use dependabot, GH actions, or renovate. 

This repo does have a GH action that should be bumping the version.
- https://github.com/elastic/elastic-agent-system-metrics/actions/workflows/bump-golang.yml
- https://github.com/elastic/elastic-agent-system-metrics/blob/main/.github/workflows/bump-golang.yml
- https://github.com/elastic/elastic-agent-system-metrics/blob/main/.github/workflows/updatecli.d/bump-golang.yml

Looking at the [comparable history in elastic-agent](https://github.com/elastic/elastic-agent/commits/main/.go-version), it looks like the GH action will bump the patch versions, but it will not bump the minor version. That part seems to be done manually.

**Review by belimawr — APPROVED** [review]

**Review by belimawr — COMMENTED** [review]
LGTM, but @rdner comment still needs to be addressed before merging.

**Review by rdner — APPROVED** [review]

**Pull request merged**